### PR TITLE
Replaces intents with combat mode

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

This PR removes intents and replaces them with a combat mode. An explanation of what this means can be found below

### Major changes:
- Disarm and Grab intents have been removed.
- Harm/Help is now combat mode, toggled by F or 4 by default
- The context/verb/popup menu now only works when you do shift+right-click
- Right click is now disarm, both in and out of combat mode.
- Grabbing is now on ctrl-click.
- If you're in combat mode, and are currently grabbing/pulling someone, and ctrl-click somewhere else, it will not release the grab (To prevent misclicks)

### Minor interaction changes:
Right click to dissasemble tables, racks, filing cabinets (When holding the right tool to do so)
Left click to stunbaton, right click to harmbaton
Right click to tip cows
Right click to malpractice surgery
Right click to hold people at gunpoint (if youre holding a gun)

## Why It's Good For The Game

Intents heavily cripple both the code and the UI design of interactions. While I understand that a lot of people will dislike this PR as they are used to intents, they are one of our weakest links in terms of explaining to players how to do specific things, and require a lot more keypresses to do compared to this.

As an example, martial arts can now be done without having to juggle 1 2 3 and 4 to switch intents quickly.

As some of you who saw the first combat mode PR, the context menu used to be disabled in combat mode. In this version it is instead on shift-right click ensuring that you can always use it in the same way.

In this version, combat mode also no longer prevents you from attacking with items when you would so before, as this was something that was commonly complained about.

The full intention of this shift in control scheme is that right click will become "secondary interaction" for items, which prevents some of the awkward juggling we have now with item modes etcetera.

## Changelog
:cl: Qustinnus
add: Intents have been replaced with a combat mode. For more info find the PR here: https://github.com/tgstation/tgstation/pull/56601
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
GOD YES

**Comment by TemporalOroboros** [comment]
Here we go again. Hopefully it gets through this time.

**Comment by Zytolg** [comment]
It's happening!

**Comment by RaveRadbury** [comment]
So you're saying that I need to enter combat mode for piggyback and fireman carry?

**Comment by Sparkezel** [comment]
I am used to intents but I want to see it live first, then I will decide.

**Comment by Acer202** [comment]
> So you're saying that I need to enter combat mode for piggyback and fireman carry?

You read my mind and posted my comment! Tactical piggybacking time.

File: code/_onclick/hud/robot.dm [hunk 1]
```diff
@@ -141,10 +141,11 @@
 	using.hud = src
 	static_inventory += using
 
-//Intent
-	action_intent = new /atom/movable/screen/act_intent/robot()
-	action_intent.icon_state = mymob.a_intent
+//Combat Mode
```
Comment (Jared-Fogle) [inline code/_onclick/hud/robot.dm:144]: indent

**Comment by Qustinnus** [comment]
at the moment, yes, I can remove the restriction on grabs from combat-mode, I do not think thats an issue.

**Comment by Pugie** [comment]
im sure this pr is good and all and i havent read the description but im leaving my thumbs down that wont change anything because i cant trust anything from shrek coder

**Comment by Jared-Fogle** [comment]
> I can remove the restriction on grabs from combat-mode, I do not think thats an issue.

i think you should

**Comment by joookschad** [comment]
im sure this pr is good and all and i havent read the description but im leaving my thumbs down that wont change anything because i cant trust anything from shrek coder

**Comment by Qustinnus** [comment]
im actually the hulk. I have low cold resistance so i'm in extremely poor health.

**Comment by LemonInTheDark** [comment]
Don't chicken out this time refactor man

**Comment by MrDoomBringer** [comment]
the shift+rightclick change to the context menu is by far the best improvement over the last attempt
qust you madman hopefully this works out

**Comment by Tlaltecuhtli** [comment]
do borgs have combat mode?

**Comment by Sparkezel** [comment]
>do borgs have combat mode?
Aren't they harm/help already? Seems like same thing tbh

**Comment by Qustinnus** [comment]
@Tlaltecuhtli code-wise yet but borg players will not notice a difference seeing as what they had was basically combat mode.

**Comment by RaveRadbury** [comment]
It seems inconsistent to make harmbaton leftclick and stunbaton right click, especially considering most of the time the baton should be primarily used for stunning?

**Comment by Qustinnus** [comment]
actually the reason its right click is because all our left clicks are harmful and right clicks would now be disarms/secondary actions.

but yeah I agree it'll definitely make some new players harmbaton on accident at first. I explain in the description how to use batons though so hopefully it'll be okay, but testmerges will show whether or not thats true.

**Comment by Pugie** [comment]
This pr is exciting

**Comment by Sparkezel** [comment]
Will you play the game after it gets merged?
image

**Comment by Qustinnus** [comment]
no

**Comment by FargothUr** [comment]
literally changing something that nobody had a problem with into a counter intuitive hotkey mess

**Comment by Qustinnus** [comment]
i originally made this to the request of 3 maintainers and re-made it to the request of the same maintainers + a bunch of people who asked me where it went.

**Comment by RaveRadbury** [comment]
image
I guess it's just because in the context of all of this (with the exception of hostage taking I guess), they are harm intents.

**Comment by FargothUr** [comment]
simply the fact you have to push more keys across the keyboard rather than 1-4, including shift clicking, ctrl clicking and now right clicking and left clicking for those other minor interactions.

no one is going to think to right click to disassemble stuff with tools, plus the left click for harmbaton is definetly a newbie trap lmao

**Comment by Qustinnus** [comment]
tell me, how do you currently dissasemble a table?

**Comment by Qustinnus** [comment]
@RaveRadbury actually all of those were not bound to harm before this except for one

**Comment by ShibaInuLord** [comment]
> 
> 
> simply the fact you have to push more keys across the keyboard rather than 1-4, including shift clicking, ctrl clicking and now right clicking and left clicking for those other minor interactions.
> 
> no one is going to think to right click to disassemble stuff with tools, plus the left click for harmbaton is definetly a newbie trap lmao
> 
> definetly wanna strain my pinky across to the CTRL button just to be able to grab people as well

i am sorry for your small pinky that cant reach ctrl

**Comment by FargothUr** [comment]
> 
> 
> tell me, how do you currently dissasemble a table?

left click out help intent with wrench/screwdriver

**Comment by ShibaInuLord** [comment]
> 
> 
> > tell me, how do you currently dissasemble a table?
> 
> left click out help intent with wrench/screwdriver

f

**Comment by Qustinnus** [comment]
no that puts it on the table and proves my point.

**Comment by FargothUr** [comment]
> 
> 
> no that puts it on the table and proves my point.

I didn't say left click in help intent

**Comment by ShibaInuLord** [comment]
> 
> 
> > no that puts it on the table and proves my point.
> 
> I didn't say left click in help intent

the problem is its only harm intent lmao

**Comment by FargothUr** [comment]
also your reasoning is mostly anecdotes of people nobody has ever heard of, and seeing the upvote/downvote ratio of the people who actually give a toss goes to show that a bunch of people doesn't mean everyone

**Comment by FargothUr** [comment]
> 
> 
> > > no that puts it on the table and proves my point.
> > 
> > 
> > I didn't say left click in help intent
> 
> the problem is its only harm intent lmao

you can use literally any other intent than help and it works

**Comment by maxymax13** [comment]
i hope this time it doesnt fuck up all your binds in the testmerge

**Comment by Qustinnus** [comment]
ah it was a typo.

either way, understanding that "dont be in help to deconstruct tables" is as clear as right-click, so I don't really see how it's more difficult to understand.

**Comment by Qustinnus** [comment]
@maxymax13 that's basically inevitable with a testmerge that removes keybindings, sorry.

**Comment by RaveRadbury** [comment]
I'm just saying please think of the new players and dumb admin brains and don't noob trap them with default harm baton. Ounce of prevention and all that.

**Comment by wesoda25** [comment]
As with the last time, I think ctrl click replacing grab is the weakest part of this pr. If you have someone grabbed, but miss their sprite with ctrl click, it breaks your grab. I think it’d be MASSIVELY better if you made it so that your grab isn’t broken if you happen to accidentally ctrl click the tile they’re on or something.

**Comment by TemporalOroboros** [comment]
> 
> 
> As with the last time, I think ctrl click replacing grab is the weakest part of this pr. If you have someone grabbed, but miss their sprite with ctrl click, it breaks your grab. I think it’d be MASSIVELY better if you made it so that your grab isn’t broken if you happen to accidentally ctrl click the tile they’re on or something.

They made ctrl-clicking on other things that can't be grabbed not break the grab.

**Comment by Qustinnus** [comment]
image

**Comment by MrDoomBringer** [comment]
As with the last time, I think ctrl click replacing grab is the weakest part of this pr. If you have someone grabbed, but miss their sprite with ctrl click, it breaks your grab. I think it’d be MASSIVELY better if you made it so that your grab isn’t broken if you happen to accidentally ctrl click the tile they’re on or something.

**Comment by Qustinnus** [comment]
@RaveRadbury 

yeah you are right, I have changed it, harmbaton is now the secondary action on stun batons.

**Comment by Angustmeta** [comment]
I think your current changes has completely removed the disarm and grab intent attacks from changeling tentacle, unless I'm reading it wrong. Those are the useful ones and should stay imo.

**Comment by Qustinnus** [comment]
yeah they're on my to-do list atm.

**Comment by Serxule** [comment]
why do you want to change tgs iconic combat, go play skyrat or citadel if you want combat mode.

**Comment by itseasytosee** [comment]
Perhaps we should move this argument to a forum? Half of the comments have been marked as spam.

**Comment by Garen7** [comment]
HOLY BASED

**Comment by Archanial** [comment]
PR of the year, great job.

**Comment by Rukofamicom** [comment]
I really really don't like the look of sound of this, but I'm also 90% certain it's just my monkey brain wanting to hate on major change.

I look forward to playing with it during testmerge.. but also wonder what happens to those of us used to making use of right click drop-down menus suddenly harming someone.

**Comment by Qustinnus** [comment]
> So you're saying that I need to enter combat mode for piggyback and fireman carry?

these restrictions have been removed btw

**Comment by ATH1909** [comment]
> > So you're saying that I need to enter combat mode for piggyback and fireman carry?
> 
> these restrictions have been removed btw

Will that lead to unintentionally trying to fireman carry someone when you just want to search their inventory?

**Comment by Jared-Fogle** [comment]
you must aggro grab them first, so hopefully not

**Comment by cacogen** [comment]
Is it being test merged? Because we can't know it won't be shit until it is.

**Comment by LemonInTheDark** [comment]
Needs a readover first I'd think, tho I don't care too much

**Comment by Fox-McCloud** [comment]
Isn't that sound toggling on/off combat mode ripped straight from Lifeweb's .rsc?

**Comment by Qustinnus** [comment]
i got it from citadel who got it from interbay and from my knowledge matt made it

**Comment by Qustinnus** [comment]
i got it from citadel who got it from interbay and from my knowledge matt made it

**Comment by jacklecroy** [comment]
Is the context menu referring to what alt clicking is now? Or is there still the stupid right click menu but it's rebound?

**Comment by Qustinnus** [comment]
the second

**Comment by CoffeeDragon16** [comment]
Is there some way to disable the combat mode toggle sound? During the testmerge a while ago I found the toggle sound to be a bit too loud, nothing too bad though. I think over time it could end up getting annoying. Or have you changed this since? I haven't seen others complain about it personally, so this could just be a me issue.

**Comment by Qustinnus** [comment]
I'm actually adding a pref for it right now. It was a more common complaint.

**Comment by PJB3005** [comment]
Better merge this now so you're used to it for when SS14 is done 😌

**Comment by SweptWasTaken** [comment]
YOU'RE COPYING SS14!!!

**Comment by PJB3005** [comment]
Shut the fuck up swept

**Comment by Naloacisamazing** [comment]
to me this looks like less of a good change and more change for the sake of change

**Comment by scarwolff** [comment]
awesome

**Comment by Jared-Fogle** [comment]
to me this looks like a good change that got merged, good work floyd. honestly happy that you managed to get this one out :+1:

**Comment by StarHorder** [comment]
so what about the menu that pops up when you right click? we just gonna ignore that and all its admin functions?

**Comment by Qustinnus** [comment]
@StarHorder shift + rightclick

**Comment by scarwolff** [comment]
> so what about the menu that pops up when you right click? we just gonna ignore that and all its admin functions?

they said it was shift click right click if you read the PR

**Comment by MrDoomBringer** [comment]
prepare for the flood of "why doesn't my right click context menu work" in ooc boys
floyd great work on this 👍

**Comment by StarHorder** [comment]
> 
> 
> @StarHorder shift + rightclick

oh. i cant read.

what about moving things around you when dragging them? are all the actions we are used to doable still?

**Comment by Jared-Fogle** [comment]
ctrl click while in combat mode

**Comment by Jared-Fogle** [comment]
or just click not like you needed to be in grab intent for that anyway

**Comment by MrDoomBringer** [comment]
> are all the actions we are used to doable still?

yes, the whole point of this PR is you can still do everything you could before it
and if for some reason that isn't the case then there will be a PR in the future to remedy it

**Comment by StarHorder** [comment]
yeah i feel dumb now. im sorry.

**Comment by ShibaInuLord** [comment]
Welcome to the new age

**Comment by CoffeeDragon16** [comment]
Great work Floyd, this'll be a great change long-term. It's definetely one of the largest and most important in a while, and I think it'll enable a lot of really cool stuff. I've wanted this for a while, and its great to see it finally in the game. Good job!

**Comment by KathrinBailey** [comment]
This is going to require a loooooot of wiki changes.
I feel somewhat sorry for who has to do that.

**Comment by LemonInTheDark** [comment]
Nother step in the right direction, glad you didn't let the players getcha down

**Comment by wesoda25** [comment]
One thing I don't like is how if you have combat mode on, you can still switch places with people

**Comment by revelation-8468** [comment]
local man plays is12 for 20 minutes and decides right click fits in regular ss13

no lol

**Comment by Akride** [comment]
Please make it so you can choose between intent mode and this one... I'm still tryna figure the TG codebase out, and now you add the most destabilizing shit I've seen on TG so far....

**Comment by Valtosin** [comment]
great

**Comment by cacogen** [comment]
I'm not sure why you wouldn't test merge this for feedback and issue reports first. It seems like a rushed change for the sake of the coders that the players then have to deal with the consequences of.

**Comment by Qustinnus** [comment]
we code for free
admins administrate for free
and you QA test for free

**Comment by cacogen** [comment]
Yeah, it's called a test merge.

**Comment by KathrinBailey** [comment]
> we code for free
> admins administrate for free
> and you QA test for free

You have a point, but let's not forget that coders and especially admins also get practical real-world job experience in babysitting and dealing with preschool children and special needs preschool children.
Oranges alone could probably be the headmaster of a Kindergarten school with his experience from being a TG maintainer.
I know that my child would be safe in Orange's school because if he just misbehaves Oranges will quickly put up a fix PR to his behaviour that removes his lunch and playtime or something.

**Comment by doguinh0** [comment]
i hope you die

**Comment by doguinh0** [comment]
> Please make it so you can choose between intent mode and this one... I'm still tryna figure the TG codebase out, and now you add the most destabilizing shit I've seen on TG so far....

please god yes

**Comment by mattroks101** [comment]
> local man plays is12 for 20 minutes and decides right click fits in regular ss13
> 
> no lol

This system doesn't work anything like IS12.

**Comment by GaryGreytide** [comment]
Everyone take a note as to who it was that merged this. 
No matter how much sugar you hopeful types mix into the diarrhea soup, it will still be liquid dookie in a bowl.
Get your collective sh*t together /tg/, there is peanut butter spaghetti falling out of your pocket.

Overall, great pr. 10/10 
Can't wait to see the people who play the game react.

**Comment by Blevruz** [comment]
i get to move my hand a little bit less 10/10 good pr

**Comment by cacogen** [comment]
I wish 1 would turn off combat the way 4 turns it on as 1 is what I used to use for help.

**Comment by itseasytosee** [comment]
You can change the keybind you know

**Comment by MrDoomBringer** [comment]
@itseasytosee no you can't, not with this PR at least.
PR #56648 adds what @cacogen wants

**Comment by Jared-Fogle** [comment]
id like for every no-label-next-to-the-smiley that asks for this to be a preference to take a look at the number of files changed. maintaining two separate systems is not viable at all, and will never happen. get used to it

**Comment by Jared-Fogle** [comment]
> Everyone take a note as to who it was that merged this.

for the record, i was the one that was going to merge it, but i said i'd merge it the day after. this pr has several :+1: from other maintainers, this would've been anyone else. you don't have the grand point you think you do

**Comment by Jared-Fogle** [comment]
> I'm not sure why you wouldn't test merge this for feedback and issue reports first. 

we did

**Comment by Escamado** [comment]
this seems good but holy fuck can we stop speed merging big game changes? I have been waiting more than a week for the botany roses pr to be merged and this gets merged in less than a day. But I guess big coders get some bonuses.

**Comment by MrDoomBringer** [comment]
@Escamado I don't see any compelling reasons to delay the merge of this. 

Having this in test-merge hell for weeks on end only serves to slow development, forcing one person (Qustinnus) to work out edge cases instead of allowing the entire codebase to iron out kinks. At the time of merging it was in a perfectly workable state for the codebase, minus some minor hotfixes. 

Also, it shuts up the droves of unconstructive feedback from people who don't like change.

**Comment by Jared-Fogle** [comment]
Also, this is a nearly 300 files changed PR. The more we delay, the more merge conflicts Floyd gets, and the higher the chance for bugs to crop up that wouldn't have otherwise been introduced.

**Comment by Felixx68** [comment]
Probably the worst change I have seen in a game, not only in what it changes but because I've never seen or heard anyone ever ask for anything even remotely like this. There was no reason to change any of this at all gameplay wise and it adds nothing. There should at least be a way to have the old intents, even if I have to do some dumb shit to use martial arts (which I liked the intents for)

**Comment by StarHorder** [comment]
this feels clunky, and the intent system is deep muscle memory for most people. im sure i'll get used to it, but it will take a while.

**Comment by KathrinBailey** [comment]
@Qustinnus Are there any plans to make stuff like the context menu and disarm look for a bind that the user can change instead of hardcoded Shift + RMB / RMB?
It'd also have the added benefits of letting ghosts use RMB for the context menu again, as they have nothing of importance bound to RMB.

I'm really looking forward to how this is going to make the game more accessible to new players though. Thanks for all the hard work on this. ❤️

**Comment by StarHorder** [comment]
yeah i... jeez this is... it feels so clunky... I dunno how i feel about this.

**Comment by DiscountDirtyDan** [comment]
Please give us the ability to opt-in to using this clientside instead of forcing everyone to use it. I have to throw out my ten years of muscle memory and knowledge of mechanics now on TG or any TG downstream. There are probably way, way better ways to approach the issue of UI confusion and lack of cohesion than just outright making everything more confusing for everyone at once, instead of just the new players.

**Comment by scrubs2009** [comment]
So let me get this straight. Most players spend most of their time during rounds not fighting. This means that for most players the majority of the time during each round they have what is basically 2 left mouse buttons?

**Comment by maxymax13** [comment]
combat mode or: how i disarmed the locker while trying to pick something out

**Comment by Jared-Fogle** [comment]
@DiscountDirtyDan @Felixx68 For the love of God please read

image

**Comment by scrubs2009** [comment]
> @DiscountDirtyDan @Felixx68 For the love of God please read
> 
> image

Then axe the shitty system all together.

**Comment by Jared-Fogle** [comment]
We just did, you're on that very PR

**Comment by scarwolff** [comment]
eyo i fink dis mane might be a genius

**Comment by DiscountDirtyDan** [comment]
> @DiscountDirtyDan @Felixx68 For the love of God please read
> 
> image

Oh don't worry, I read it. That doesn't mean I'm going to suddenly pretend this change is good. The system before controlled perfectly fine. There is probably a way better way to fix what you deem a convoluted control system than making it *more* convoluted.

**Comment by Jared-Fogle** [comment]
PRs are always open, feel free to ask in #coding-general how to get started.

**Comment by scrubs2009** [comment]
PR being put on my to do list. Rollback this one
ez fix.

**Comment by toomykins** [comment]
i'll see you all on /vg/

**Comment by actioninja** [comment]
@toomykins bye!

**Comment by actioninja** [comment]
@KathrinBailey
If you want the context menu to be rebindable, go complain to Lummox. We literally cannot change what button it is on period. It's either on raw right click or modifier right click, no other options.

As for making things rebindable or allowing you to flip the default (right click is context menu, shift right click is alt action,) then we would run the risk of a large portion of people running non-default controls and future assumptions about how things work being cloudy. "Just change X control this is better" is a trap, because the end goal here is better integrating alternate usage of items across the whole game. Relegating that alternate usage to shift+right click would result in endless situations of "how do I do x thing right click just opens menu" and having to account for both control schemes in every single design decision.
It's not as simple as something like just allowing discrete bindings to be made because it represents serious and significant control layout changes, and how the game fundamentally controls is something we should try to curate on a default level. Rebinds should always be to allow experienced players to change their layout to a more familiar one, or to let people with non-qwerty layouts actually have sane control schemes. I don't think we should be catering specifically to people that are going to complain their tightly refined muscle memory for the highly competitive team deathmatch game ss13 is going to be affected resulting in 50 lower APM for their next 20 mime toolboxings.
tl;dr: controls are likely going to shift over time and designing around controlled defaults is a very useful paradigm. Given the chance, we have a non zero amount of players that will absolutely wreck their control scheme and then complain their control scheme is wrecked.

**Comment by KathrinBailey** [comment]
> @KathrinBailey
> If you want the context menu to be rebindable, go complain to Lummox. We literally cannot change what button it is on period. It's either on raw right click or modifier right click, no other options.

Never would've guessed that the context menu in that way, huh.
That's perfectly fair and a good reason, personally I just wanted to be able to swap the context menu back to RMB and disarm (or the 'special action') to Shift + RMB clientside, because I personally find myself using the context menu way more than disarm (or the special cool button) with ghosting and answering the gazillions of tickets. But I'll get used to it.

Thanks for taking the time to respond in the sea of brigading and salt comments though. ❤️

**Comment by Daelso** [comment]
holy based change!!!!

**Comment by cacogen** [comment]
Still doesn't make sense that right-click with combat mode off isn't the context menu instead of disarm.

**Comment by MrDoomBringer** [comment]
> Still doesn't make sense that right-click with combat mode off isn't the context menu instead of disarm. You can keep shift+right-click as the context menu there too to avoid confusion.

this would just make the whole thing way more confusing.

**Comment by Felixx68** [comment]
> image
> 
> We just did, you're on that very PR

not quite

**Comment by itseasytosee** [comment]
I'd like to restate my proposal of locking the PR and making a form discussion. Or hell just locking the PR.

**Comment by cacogen** [comment]
>this would just make the whole thing way more confusing.

No, letting right-click be normal without combat mode would be one less hurdle for people to both learn and memorise and allowing the shift right-click context menu outside of combat mode would prevent people doing the wrong thing after switching out of it. It's more intuitive if switching to combat mode affects right-click's functionality than if it's changed regardless.

**Comment by Al-1ce** [comment]
>most of the community doesn't want this
>merged by oranges
Yep, /tg/ alright.

**Comment by cacogen** [comment]
Now that we're stuck with it, it could be good with some tweaking.

**Comment by TrevinBehrens** [comment]
Why for the love of god this has made my friends learning the game have a terrible experience. Not only is it harder to swap codebases but it's also just as bad as intents.

**Comment by Jared-Fogle** [comment]
why are you teaching your friends the game by swapping around entirely different codebases. i think you are a bad teacher

**Comment by Jared-Fogle** [comment]
ok ive had my lulz, take future feedback to the forums: https://tgstation13.org/phpBB/viewtopic.php?f=10&t=28430

we're already building features on top of this, this is non negotiable

**Comment by optimumtact** [comment]
not fair I'm banned from the forum how am I supposed to contribute

**Comment by Mothblocks** [comment]
image
